scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,189 @@
+from typing import Optional
+
+# Workaround to use the preview SDK
+from azure.search.documents.indexes._generated.models import (
+    IndexProjectionMode,
+    SearchIndexerIndexProjections,
+    SearchIndexerIndexProjectionSelector,
+    SearchIndexerIndexProjectionsParameters,
+    SearchIndexerSkillset,
+)
+from azure.search.documents.indexes.models import (
+    AzureOpenAIEmbeddingSkill,
+    AzureOpenAIParameters,
+    AzureOpenAIVectorizer,
+    FieldMapping,
+    InputFieldMappingEntry,
+    OutputFieldMappingEntry,
+    SearchIndexer,
+    SearchIndexerDataContainer,
+    SearchIndexerDataSourceConnection,
+    SplitSkill,
+)
+
+from .blobmanager import BlobManager
+from .embeddings import AzureOpenAIEmbeddingService
+from .listfilestrategy import ListFileStrategy
+from .searchmanager import SearchManager
+from .strategy import DocumentAction, SearchInfo, Strategy
+
+
+class IntegratedVectorizerStrategy(Strategy):
+    """
+    Strategy for ingesting and vectorizing documents into a search service from files stored storage account
+    """
+
+    def __init__(
+        self,
+        list_file_strategy: ListFileStrategy,
+        blob_manager: BlobManager,
+        embeddings: Optional[AzureOpenAIEmbeddingService],
+        subscriptionId: str,
+        searchServiceUserAssginedId: str,
+        document_action: DocumentAction = DocumentAction.Add,
+        search_analyzer_name: Optional[str] = None,
+        use_acls: bool = False,
+        category: Optional[str] = None,
+    ):
+        self.list_file_strategy = list_file_strategy
+        self.blob_manager = blob_manager
+        self.document_action = document_action
+        self.embeddings = embeddings
+        self.subscriptionId = subscriptionId
+        self.search_user_assigned_identity = searchServiceUserAssginedId
+        self.search_analyzer_name = search_analyzer_name
+        self.use_acls = use_acls
+        self.category = category
+
+    async def createEmbeddingSkill(self, index_name: str):
+        skillset_name = f"{index_name}-skillset"
+
+        split_skill = SplitSkill(
+            description="Split skill to chunk documents",
+            text_split_mode="pages",
+            context="/document",
+            maximum_page_length=2048,
+            page_overlap_length=20,
+            inputs=[
+                InputFieldMappingEntry(name="text", source="/document/content"),
+            ],
+            outputs=[OutputFieldMappingEntry(name="textItems", target_name="pages")],
+        )
+
+        if self.embeddings is None:
+            raise ValueError("Expecting Azure Open AI instance")
+
+        embedding_skill = AzureOpenAIEmbeddingSkill(
+            description="Skill to generate embeddings via Azure OpenAI",
+            context="/document/pages/*",
+            resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+            deployment_id=self.embeddings.open_ai_deployment,
+            auth_identity=None,
```

Comment:
When would auth_identity be needed?

Reply:
Only when then plan to use user assigned managed identity instead of system assigned. By default when left with None, search service resolves on managed identity.